### PR TITLE
Cs/show wfs layer with XML query

### PR DESCRIPTION
**Pull request by changanxian**
Adds a new sample to show a WFS layer with custom XML query

### CODE REVIEW

File: lib/samples/show_wfs_layer_with_xml_query/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+`QueryParameters` objects can't represent all possible queries that can be made against a WFS feature service. For example, query parameters don't support wildcard searches. You can provide queries as raw XML strings, allowing you to access query functionality not available with `QueryParameters`.
+
+## How to use the sample
+
+Run the sample and view the data loaded from the WFS feature table.
+
+## How it works
+
+1. Create a `WfsFeatureTable` and a `FeatureLayer` to visualize the table.
+2. Set the feature request mode to `ManualCache`.
```
Comment (PaulAllanSturm) [inline lib/samples/show_wfs_layer_with_xml_query/README.md:18]: I think for Flutter this should say `FeatureRequestMode.manualCache`.

File: lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+void main() async {
+  // Supply your apiKey using the --dart-define-from-file command line argument
+  ArcGISEnvironment.apiKey = const String.fromEnvironment('API_KEY');
+
+  runApp(
+    const MaterialApp(
+      home: ShowWfsLayerWithXmlQuery(),
+    ),
+  );
+}
```
Comment (PaulAllanSturm) [inline lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:29]: The `main()` function should be deleted from individual samples.

File: lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+class ShowWfsLayerWithXmlQuery extends StatefulWidget {
+  const ShowWfsLayerWithXmlQuery({super.key});
+
+  @override
+  State<ShowWfsLayerWithXmlQuery> createState() =>
+      _ShowWfsLayerWithXmlQueryState();
+}
+
+class _ShowWfsLayerWithXmlQueryState extends State<ShowWfsLayerWithXmlQuery> {
```
Comment (PaulAllanSturm) [inline lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:39]: This class should mix in `with SampleStateSupport`

File: lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
```
Comment (PaulAllanSturm) [inline lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:48]: We don't need a SafeArea when the sample is just a MapView without any controls. For example, see display_annotation.dart.

File: lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  Future<void> loadWfsLayerWithXmlQuery() async {
+    const wfsFeatureTableUri =
+        'https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&amp;request=getcapabilities';
+
+    // create the WFS feature table from URI and name
```
Comment (PaulAllanSturm) [inline lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:94]: Comments should be capitalized with a period at the end.

File: lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISNavigation);
```
Comment (PaulAllanSturm) [inline lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:68]: One last thing -- we would normally put a comment above this line,
```dart
    // Create a map with the ArcGIS Navigation basemap style and set to the map view.
```

**Comment by jenmerritt** [comment]
We need the `generated_samples_list.json` updating

**Comment by jenmerritt** [comment]
Otherwise looks good!

File: lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+      uri: Uri.parse(wfsFeatureTableUri),
+      tableName: 'Seattle_Downtown_Features:Trees',
+    )
+      // Set the feature request mode and axis order.
+      ..axisOrder = OgcAxisOrder.noSwap
+      ..featureRequestMode = FeatureRequestMode.manualCache;
+
+    // Create the feature layer from the feature table.
+    final featureLayer = FeatureLayer.withFeatureTable(statesTable);
+    await featureLayer.load();
```
Comment (kmueller-gis) [inline lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:93]: Do you need to actively load the layer, or would adding it to the map's operationalLayers be enough?

Comment (changanxian) [reply lib/samples/show_wfs_layer_with_xml_query/show_wfs_layer_with_xml_query.dart:94]: we load here specifically to instantiate the layer to ensure the query and full extent correctly.

**Review by kmueller-gis — APPROVED** [review]

**Review by PaulAllanSturm — APPROVED** [review]

**Pull request merged**